tests/test_x509.py

```diff
@@ -1420,6 +1421,49 @@ def test_build_cert(self, backend):
             x509.DNSName(u"cryptography.io"),
         ]
 
+    def test_build_cert_random_serial(self, backend):
+        sample_data = os.urandom(20)
+
+        @contextmanager
```

Comment:
Can't. That will patch from whenever you call it until end of the test, and since OpenSSL consumes randomness from `os.urandom` due to our random engine that means it will call the monkeypatched version when we don't want it to.